compiler/plugins/target/AMD-AIE/iree-amd-aie/IR/AMDAIEDialect.cpp

```diff
@@ -22,9 +26,126 @@ struct AMDAIEDialectOpAsmInterface : public OpAsmDialectInterface {
+                         DialectAsmParser &parser) {
+  if (OptionalParseResult parseResult = amdaieTypeParser(parser, name, result);
+      parseResult.has_value())
+    return parseResult.value();
+
+  parser.emitError(parser.getNameLoc(), "unknown AMDAIE dialect type: \"")
+      << name << "\"";
+  return failure();
+}
+
+/// Parse an instance of a type registered to the AMDAIE dialect.
+Type AMDAIEDialect::parseType(DialectAsmParser &parser) const {
+  StringRef name;
+  Type result;
+  if (parser.parseKeyword(&name) || parse(result, name, parser)) return {};
+  return result;
+}
+
+/// Print an instance of a type registered to the AMDAIE dialect.
+void AMDAIEDialect::printType(Type type, DialectAsmPrinter &printer) const {
+  if (type.isa<AMDAIELogicalObjectFifoType>()) {
+    auto objectFifoType = type.cast<AMDAIELogicalObjectFifoType>();
+    printer << "logicalobjectfifo<";
+    printer << objectFifoType.getElementType();
+    printer << '>';
+  }
+}
+
 void AMDAIEDialect::initialize() {
+  addTypes<AMDAIELogicalObjectFifoType>();
   initializeAMDAIEAttrs();
+  initializeAMDAIEOps();
   addInterfaces<AMDAIEDialectOpAsmInterface>();
 }
 
+namespace detail {
+/// This class represents the internal storage of the AMDAIE
+/// `LogicalObjectFifoType`.
+struct AMDAIELogicalObjectFifoTypeStorage : TypeStorage {
+  /// The `KeyTy` is a required type that provides an interface for the storage
+  /// instance. This type will be used when uniquing an instance of the type
+  /// storage.
+  using KeyTy = MemRefType;
+
+  /// A constructor for the objectFifo type storage instance.
+  AMDAIELogicalObjectFifoTypeStorage(MemRefType elementType)
+      : elementType(elementType) {}
+
+  /// Define the comparison function for the key type with the current storage
+  /// instance. This is used when constructing a new instance to ensure that we
+  /// haven't already uniqued an instance of the given key.
+  bool operator==(const KeyTy &key) const { return key == KeyTy(elementType); }
+
+  /// Define a construction method for creating a new instance of this storage.
+  /// This method takes an instance of a storage allocator, and an instance of a
+  /// `KeyTy`.
+  static AMDAIELogicalObjectFifoTypeStorage *construct(
+      TypeStorageAllocator &allocator, const KeyTy &key) {
+    // Allocate the storage instance and construct it.
+    return new (allocator.allocate<AMDAIELogicalObjectFifoTypeStorage>())
+        AMDAIELogicalObjectFifoTypeStorage(key);
+  }
+
+  MemRefType elementType;
+};
+}  // namespace detail
+
+AMDAIELogicalObjectFifoType AMDAIELogicalObjectFifoType::get(
+    MemRefType elementType) {
+  // Call into a helper 'get' method in 'TypeBase' to get an uniqued instance
+  // of this type.
+  MLIRContext *ctx = elementType.getContext();
+  return Base::get(ctx, elementType);
+}
+
+LogicalResult AMDAIELogicalObjectFifoType::verify(
+    function_ref<InFlightDiagnostic()> emitError, MemRefType elementType) {
+  return success();
```

Comment:
do we want to do some actual verification here, e.g fail on dynamic shaped memrefs?

Reply:
Yes, dynamic shapes might be a good one to check. Will add it.